### ISSUE TITLE
Report on time per stage

### ISSUE THREAD

**Issue by michaelvirnstein** [cobaltdev/pipeline#3]
It would be great if i could get a visual report of the time each stage took over a periode of builds. This way it would be easy to identify bottlenecks and see if any improvements are really working

**Comment by brmakana** [comment]
Good idea! We will add this to the backlog and see about getting it in an upcoming version.

**Comment by bessieyy** [comment]
Really a good idea! But still need to think about what the duration of a stage is. All jobs within a stage can be processed at the same time with multiple agents, so it may not be the best choice to use the sum of jobs durations. It's not reasonable to use the duration between the start and end date time of a stage either. Jobs can be done independently and they may be blocked by jobs from other plans that use up all the build agents. If there're more plans running together, the stage duration may be longer, it's hard to tell the performance.

But it's really a great idea, just need to consider more to include it in next version.

**Comment by dougmessick** [comment]
Thank you for the feedback, and I agree with you about time calculations. We will add it to the backlog.

**Comment by michaelvirnstein** [comment]
> Really a good idea! But still need to think about what the duration of a stage is. All jobs within a stage can be processed at the same time with multiple agents, so it may not be the best choice to use the sum of jobs durations. It's not reasonable to use the duration between the start and end date time of a stage either. Jobs can be done independently and they may be blocked by jobs from other plans that use up all the build agents. If there're more plans running together, the stage duration may be longer, it's hard to tell the performance.

That is true. I just thought that it would be an easy start and it is always possible to add more information later or even create a possibility to have some sort of drill down. Like i can see the time each stage has been taking and then i can drill down into each stage and see the time each individual job has been taking. From there I could be possible to drill down into the job and see e.g. time waiting for free agent, time each task was taking etc. 

But for the start, I think that it already helps to have some basic numbers.
